drivers/pure/flasharray/fa.go

```diff
@@ -128,6 +130,12 @@ func (c *Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 		return nil, err
 	}
 	defer resp.Body.Close()
+	bodyBytes, _ := ioutil.ReadAll(resp.Body)
+	bodyString := string(bodyBytes)
+	// This is for Deletepod where we are returning nil if we are retention-lock  which is expected error
```

Comment:
[nit] Can we also mention "Safe Mode" somewhere in this comment? That's the magic word to let people know why this is happening, "the array has safe mode enabled"